### PR TITLE
Adds viro tape

### PR DESCRIPTION
**Pull request by Apogee-dev**
virotape
Adds biohazard tape, used to cordon off infected areas. Medbay staff can pass through without breaking it, and it can be found in biohazard lockers in medbay storage and virology. The syndicate version, available at a discount to the CMO and Virologist, throws unstable goo at anyone that attempts to break it without wearing latex gloves.

:cl:
 * rscadd: Added biohazard tape, found in biohazard lockers in medbay and viro. Can be lifted by medical staff.
 * rscadd: Added syndicate biohazard tape, which throws unstable goo if you attempt to break it without latex gloves.

### CODE REVIEW

**Comment by DeityLink** [comment]
Haha, YES!

File: code/game/objects/items/policetape.dm [hunk 1]
```diff
@@ -397,3 +418,29 @@
...
+
+// Viro syndie tape : Sturdy, throws unstable goo when you try to break it
+
+/obj/item/tape/viro/syndie/destroy_tape(var/mob/user, var/obj/item/weapon/W)
+
+	if (istype(user, /mob/living))
+		var/mob/living/L = user
+		if (ishuman(user))
+			var/mob/living/carbon/human/H = user
+			var/obj/item/clothing/gloves/G = H.get_item_by_slot(slot_gloves)
```
Comment (DamianX) [inline code/game/objects/items/policetape.dm]: use is_wearing_item and save two lines of code

**Comment by BomberBro** [comment]
Don't particularly like the concept of being able to pass through tape innately like cult doors because your ID has medical access, it should work like all the others. This is just theorycrafting, but as a doctor or (more likely) paramedic you could basically never get caught by security or el antag because you can just mindlessly throw down tape they can't walk through, not to mention that in a truly Deffcon 1 scenario you could do a lot worse with an infinite weak wall that doesn't slow you down.

Also, the justification is kind of weird, how often do biohazardous areas need tape? Engineering gets it because breaches are common, but in this case I don't see why there needs to be tape for medbay, especially tape that's better than every other tape in the game for some reason.

**Comment by DeityLink** [comment]
Can we see pics of it?

File: vgstation13.dme [hunk 2]
```diff
@@ -2756,7 +2756,7 @@
 #include "maprendering\maprendering.dm"
 #include "maps\_map.dm"
 #include "maps\_map_override.dm"
-#include "maps\lowfatbagel.dm"
+#include "maps\tgstation.dm"
```
Comment (Exxion) [inline vgstation13.dme]: Huh? Why was this changed, but also why was it lowfatbagel before?

**Comment by Apogee-dev** [comment]
> Don't particularly like the concept of being able to pass through tape innately like cult doors because your ID has medical access, it should work like all the others. This is just theorycrafting, but as a doctor or (more likely) paramedic you could basically never get caught by security or el antag because you can just mindlessly throw down tape they can't walk through, not to mention that in a truly Deffcon 1 scenario you could do a lot worse with an infinite weak wall that doesn't slow you down.

I mean, not that I disagree with you on that point, but that's an issue with tape in general since virotape behaves exactly like the other three types with the exception of a different access requirement.

> Also, the justification is kind of weird, how often do biohazardous areas need tape? Engineering gets it because breaches are common, but in this case I don't see why there needs to be tape for medbay, especially tape that's better than every other tape in the game for some reason.

I'll be honest, it's probably mostly going to be used to tape off the doors to virology, I'm not sure what you mean by "better than every other tape in the game," though, unless you're referring to the syndicate version. The mundane version is identical to all other tapes. I can easily tone down the damage on the syndietape by adjusting throw speed if necessary.

**Comment by Eneocho** [comment]
`>be antag`
`>place some viro tape somewhere`
`>grab a pair of insulos because doors are shocked and >using latex instead of insulos`
`>pass by the tape`
`>forget I was wearing insulos instead of latex`
`>get goo'd`

**Comment by Apogee-dev** [comment]
> 
> 
> `>be antag CMO`
> `>place some viro tape somewhere`
> `>grab a pair of insulos because doors are shocked`
> `>pass by the tape`
> `>forget I was wearing insulos instead of latex`
> `>get goo'd`

Antag tape still lets the person who placed it lift it with no problem, but you'll get goo'd if you try to tear it down.

**Comment by Eneocho** [comment]
That's fair.

File: vgstation13.dme (hunk 2 above)
Comment (Apogee-dev) [reply vgstation13.dme]: Unless I'm adding new files I usually don't even pay attention to the dme honestly

**Comment by gurfan** [comment]
> Don't particularly like the concept of being able to pass through tape innately like cult doors because your ID has medical access, it should work like all the others. This is just theorycrafting, but as a doctor or (more likely) paramedic you could basically never get caught by security or el antag because you can just mindlessly throw down tape they can't walk through, not to mention that in a truly Deffcon 1 scenario you could do a lot worse with an infinite weak wall that doesn't slow you down.

Cant all of this be said for security tape too?

**Comment by Eneocho** [comment]
Still, Goo is kinda strong for what it's worth. Engitape shocks you, atmostape damages or cuts your hands and seccietape instacuffs you, but none of these punishments are inherently deadly, you can even cut down the tapes no problem with something sharp. Goo is (the few times I've seen it in use) very, very strong, to the point of being able to instakill on one throw. Not being able to remove the tape as the other tapes and getting instakilled is both a consistency issue and very overpowered.

**Comment by Apogee-dev** [comment]
> 
> 
> Still, Goo is kinda strong for what it's worth. Engitape shocks you, atmostape damages or cuts your hands and seccietape instacuffs you, but the three you can cut down no problem with some wirecutters. Goo is (the few times I've seen it in use) very, very strong, to the point of being able to instakill on one throw. Not being able to remove the tape as the other tapes is both a consistency issue and very overpowered.

Like all thrown objects, goo's damage scales with throw speed, and the tape doesn't throw it as hard as a human. Right now, it does 30-40 damage to an unprotected spaceman, with a chance to cause IB. Also, the cutting behavior is not unique. Engineering tape still tries to electrocute you no matter what you use on it, the only protection is glubbs. Viro tape works the same way, just with latex gloves, which are much easier to find than insulated gloves.

For reference, atmos and sec tape both require a sharp object with a force greater than or equal to 10 to break. Engineering tape requires any sharp object, but electrocutes you if you aren't wearing glubbs. Viro tape can actually be broken empty-handed, but throws goo at you if you aren't wearing latex gloves. It's roughly comparable to engitape in that regard.

**Comment by Apogee-dev** [comment]
> Still, Goo is kinda strong for what it's worth. Engitape shocks you, atmostape damages or cuts your hands and seccietape instacuffs you, but the three you can cut down no problem with some wirecutters. Goo is (the few times I've seen it in use) very, very strong, to the point of being able to instakill on one throw. Not being able to remove the tape as the other tapes is both a consistency issue and very overpowered.

Like all thrown objects, goo's damage scales with throw speed. The tape actually doesn't throw it as hard as a human and I can (and probably will) tone it down further. At the moment the damage it does is highly variable but the most I've seen is 60. Also, viro tape isn't the only one that still slaps you if you try to use a sharp object on it: Engineering tape electrocutes you regardless of what you're trying to cut it with, as long as you aren't wearing glubbs. Viro tape behaves the same way, except you can break it without a sharp tool, and latex gloves are much easier to get than insulated gloves.

**Comment by BomberBro** [comment]
> > Don't particularly like the concept of being able to pass through tape innately like cult doors because your ID has medical access, it should work like all the others. This is just theorycrafting, but as a doctor or (more likely) paramedic you could basically never get caught by security or el antag because you can just mindlessly throw down tape they can't walk through, not to mention that in a truly Deffcon 1 scenario you could do a lot worse with an infinite weak wall that doesn't slow you down.
> 
> Cant all of this be said for security tape too?

Security tape can't be phased through by security. If that's not how the tape works and you can't just innately vibrate through it while it's up then ignore my comment entirely, I asked a collab about the code since I wasn't sure and they gave me the impression that's how it worked, but I didn't fully understand it to begin with.

**Comment by Apogee-dev** [comment]
> Security tape can't be phased through by security. If that's not how the tape works and you can't just innately vibrate through it while it's up then ignore my comment entirely, I asked a collab about the code since I wasn't sure and they gave me the impression that's how it worked, but I didn't fully understand it to begin with.

That's not how it works, no. The access requirement is for which jobs are able to lift the tape to allow passage through it, just like sec tape. Anyone else will attempt to rip the whole line down. The basic viro tape has absolutely no unique code and functions identically to every other form of tape.
